pallets/runtime/src/asset.rs

```diff
@@ -435,26 +440,10 @@ decl_module! {
 
             ensure!(total_supply <= MAX_SUPPLY.into(), Error::<T>::TotalSupplyAboveLimit);
 
-            // Alternative way to take a fee - fee is proportionaly paid to the validators and dust is burned
-            let validators = <pallet_session::Module<T>>::validators();
-            let fee = Self::asset_creation_fee();
-            let validator_len:T::Balance;
-            if validators.is_empty() {
-                validator_len = T::Balance::from(1 as u32);
-            } else {
-                validator_len = T::Balance::from(validators.len() as u32);
-            }
-            let proportional_fee = fee / validator_len;
-            // for v in validators {
-            //     <balances::Module<T> as Currency<_>>::transfer(
-            //         &sender,
-            //         &<T as Utils>::validator_id_to_account_id(v),
-            //         proportional_fee,
-            //         ExistenceRequirement::AllowDeath
-            //     )?;
-            // }
-            let remainder_fee = fee - (proportional_fee * validator_len);
-            let _withdraw_result = <balances::Module<T>>::withdraw(&sender, remainder_fee, WithdrawReason::Fee.into(), ExistenceRequirement::KeepAlive)?;
+            <<T as IdentityTrait>::ProtocolFee>::charge_fee(
+                &signer,
+                ProtocolOp::AssetCreateToken,
+            )?;
             <identity::Module<T>>::register_asset_did(&ticker)?;
 
             if is_ticker_available_or_registered_to == TickerRegistrationStatus::Available {
```

Comment:
We need to also charge the ticker registration fee if a ticker is being registered along with token registration (L451).

Reply:
If acceptable, I'd charge the two fees in two separate operations in this PR. We can introduce a sum of protocol operations in another PR later. This is the first instance where it is needed.

Reply:
Yeah, that's fine for the testnet v1 at least.

Reply:
Done. :smiley: